// allure/dist/commands/projects/create.js
import { getGitRepoName, readConfig } from "@allurereport/core";
import { AllureServiceClient, KnownError } from "@allurereport/service";
import prompts from "prompts";
import { green, red } from "yoctocolors";
import { createCommand } from "../../utils/commands.js";
import { logError } from "../../utils/logs.js";
export const ProjectsCreateCommandAction = async (projectName, options) => {
    const { config: configPath, cwd } = options ?? {};
    const config = await readConfig(cwd, configPath);
    if (!config?.allureService?.url) {
        console.error(red("No Allure Service URL is provided. Please provide it in the `allureService.url` field in the `allure.config.js` file"));
        process.exit(1);
        return;
    }
    const serviceClient = new AllureServiceClient(config.allureService);
    let name = projectName;
    if (!name) {
        try {
            name = await getGitRepoName();
        }
        catch (ignored) { }
    }
    if (!name) {
        const res = await prompts({
            type: "text",
            name: "name",
            message: "Enter project name",
        });
        name = res?.name;
    }
    if (!name) {
        console.error(red("No project name provided!"));
        process.exit(1);
        return;
    }
    try {
        const project = await serviceClient.createProject({
            name,
        });
        const lines = [
            `The "${green(project.name)}" has been created. Insert following code into your Allure Config file, to enable Allure Service features for the project:`,
            "",
            green("{"),
            green("  allureService: {"),
            green(`    project: "${project.name}"`),
            green("  }"),
            green("}"),
        ];
        console.info(lines.join("\n"));
    }
    catch (error) {
        if (error instanceof KnownError) {
            console.error(red(error.message));
            process.exit(1);
            return;
        }
        await logError("Failed to create project due to unexpected error", error);
        process.exit(1);
    }
};
export const ProjectsCreateCommand = createCommand({
    name: "project-create [name]",
    description: "",
    options: [
        [
            "--config, -c <file>",
            {
                description: "The path Allure config file",
            },
        ],
        [
            "--cwd <cwd>",
            {
                description: "The working directory for the command to run (default: current working directory)",
            },
        ],
    ],
    action: ProjectsCreateCommandAction,
});
